### ISSUE TITLE
Refactor server request calling logic in views

### ISSUE THREAD

**Issue by sderickson** [codecombat/codecombat#3712]
Implement these recipes across the site:
* [Add methods for special server endpoints](https://github.com/codecombat/codecombat/wiki/Add-methods-for-special-server-endpoints)
* [Call server from view](https://github.com/codecombat/codecombat/wiki/Call-server-from-view)

Right now there are a lot of instances of making a model or CocoCollection, manipulating the parameters and url directly, then handing it to supermodel's `loadModel` or `loadCollection` methods. Instead, methods should be added to models and collections according to the first recipe and used according to the second.

**Comment by ogennadi** [comment]
@sderickson, my understanding is that I need to refactor all locations where loadModel or loadCollection is called using a "hand-modified" URL. Are these all the locations that need refactoring?

(Some loadModel calls seem to just set an id before calling fetch. They look too simple to require a refactoring.)

Example URL-using call
https://github.com/codecombat/codecombat/blob/5bc612c0eaf2013d7f193907cdba01a20c57a642/app/views/play/level/modal/PicoCTFVictoryModal.coffee#L21


Example id-using call
https://github.com/codecombat/codecombat/blob/ea7b94bba13c8b8c96c06e2fad71149653972a1b/app/views/common/UserView.coffee#L20

**Comment by sderickson** [comment]
Yes, the main goal is to remove all "hand-modified" urls, so that urls don't appear at all in view logic. That should be encapsulated in models or (going forward), fetch api-style functions: https://github.com/codecombat/codecombat/tree/master/app/core/api

I recommend also refactoring those id ones to use `trackRequest`. So the second link would become:

```
@user = new User _id: @userID
@supermodel.trackRequest @user.fetch(cache: false)
```
